### PR TITLE
BUG,TST: Remove the misguided `run_command` that wraps subprocess

### PR DESCRIPTION
**Pull request by eric-wieser**
Also switches to using a parametrized test, for better error messages

Fixes the windows problem I reported on #12583

### CODE REVIEW

File: numpy/tests/test_scripts.py [hunk 1]
```diff
@@ -17,74 +17,13 @@
...
-    ------
-    RuntimeError
-        If `check_code` is True, and return code !=0
-    """
-    cmd = [cmd] if isinstance(cmd, basestring) else list(cmd)
-    if os.name == 'nt':
-        # Quote any arguments with spaces. The quotes delimit the arguments
-        # on Windows, and the arguments might be file paths with spaces.
-        # On Unix the list elements are each separate arguments.
-        cmd = ['"{0}"'.format(c) if ' ' in c else c for c in cmd]
```
Comment (eric-wieser) [inline numpy/tests/test_scripts.py]: This was plain wrong

**Comment by eric-wieser** [comment]
Code coverage shows the `xfail`ing test not having coverage.

That test is generally broken unless you're in a virtualenv / running tests against your system-installed numpy.

However, the new test added in #12582 passes for me.

**Comment by mattip** [comment]
LGTM, this is much simpler

**Comment by mattip** [comment]
Thanks Eric

**Pull request merged**